rpc/src/rpc_types.rs

```diff
@@ -283,19 +283,42 @@ pub struct TransactionRecipient {
     pub amount: u64,
 }
 
-#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+/// The crawled known network and measurements.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct NetworkGraph {
-    pub vertices: HashSet<Vertice>,
-    pub edges: HashSet<Edge>,
+    /// The number of nodes in the known network.
+    pub node_count: usize,
+    /// The number of connections in the known network.
+    pub connection_count: usize,
+    /// The density of the network: actual connections divided by the number of possible
+    /// connections.
+    pub density: f64,
+    /// The fiedler eigenvalue.
+    pub algebraic_connectivity: f64,
+    /// The difference between the node with the largest connection count and the node with the
+    /// lowest.
+    pub degree_centrality_delta: f64,
+
+    /// Known nodes.
+    pub vertices: Vec<Vertice>,
+    /// Known connections.
+    pub edges: Vec<Edge>,
 }
 
-#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
+/// Metadata and measurements pertaining to a node in the graph of the known network.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct Vertice {
     pub addr: SocketAddr,
     pub is_bootnode: bool,
```

Comment:
this is something I missed before: this doesn't necessarily determine whether the node is actually a bootnode or not, only whether it's a known/curated bootnode, as it's based on our own list of bootnodes; this distinction might not matter right now, but it might in the future, so I'm just pointing it out :detective:

Reply:
Indeed, we don't actually currently communicate that information through the network protocol.